pkg/core/function.go

```diff
@@ -43,28 +43,31 @@ import (
 )
 
 const (
-	functionLabel   = "riff.projectriff.io/function"
-	buildAnnotation = "riff.projectriff.io/nonce"
-	// annotation names with slashes are rejected :-/
+	functionLabel                 = "riff.projectriff.io/function"
+	buildAnnotation               = "riff.projectriff.io/nonce"
 	buildpackBuildImageAnnotation = "riff.projectriff.io-buildpack-buildImage"
 	buildpackRunImageAnnotation   = "riff.projectriff.io-buildpack-runImage"
+	functionArtifactAnnotation    = "riff.projectriff.io/artifact"
```

Comment:
The format of this annotation is inconsistent with the format of `buildpackRunImageAnnotation`. The rationale for the latter format is probably due to the belief in the deleted comment that "annotation names with slashes are rejected", but then how can `buildAnnotation` contain a slash? Since we are still in Alpha, I think it would make sense to switch the other annotation names to use slashes so the whole lot are consistent. Happy to defer that to a separate issue if you want to avoid scope creep in this PR.

Reply:
Yes, came to the same conclusion. Indeed slashes are accepted for me.
Also think it's fine to change the old ones, in another PR

Reply:
I've raised issue https://github.com/projectriff/riff/issues/1076 for now as any PR would doubtless clash with this one and there's no point _making_ work.